src/control/destination/resolution.rs

```diff
@@ -282,30 +278,28 @@ impl Updater {
         Ok(())
     }
 
-    fn no_endpoints(&mut self, exists: bool) -> Result<(), ()> {
-        if !exists {
-            // If the endpoint doesn't exist in service discovery, indicate that
-            // we should fall back.
-            trace!("does not exist in service discovery");
-            self.send(Update::NoEndpoints)?;
-        }
-
+    fn remove_all(&mut self, reason: &'static str) -> Result<(), ()> {
         for addr in self.seen.drain(..) {
-            trace!("remove {} (no endpoints)", addr);
+            trace!("remove {} ({})", addr, reason);
             self.tx
                 .unbounded_send(Update::Remove(addr))
                 .map_err(|_| ())?;
         }
         Ok(())
     }
+
+    fn does_not_exist(&mut self) -> Result<(), ()> {
```

Comment:
Hmmm, should this be called something else to distinguish from the other not-exists cases? Perhaps `unresolveable` or something like that?

Reply:
changed in fe9d612a8d4f48494a17a776cc80757c52d0f834, good call